### PR TITLE
Why DevSecOps Need Security as Code

### LINKED ISSUE #5618: Why DevSecOps Need Security as Code
#### Proposed title of article
Why DevSecOps Need Security as Code

#### Proposed article introduction
DevOps was introduced to make the software development life cycle shorter. While software development organizations breathed a sigh of relief with DevOps making the process smoother, there was another challenge. Implementing security in the software became difficult. And to solve this issue, DevSecOps was introduced.

Security as code is key to achieving DevSecOps. It requires that security be defined at the beginning of a project and codified for repeated and consistent use. In this way, it gives developers a self-service option for ensuring their code is secure.

#### Key takeaways 

1. What is security as code and DevSecOps?
2. Continuous delivery as a foundation of security as code.
3. How security as code enables development teams to address security.
4. How security as code addresses risks in development and runtime.

### PR DESCRIPTION
**Pull request by nellyat**
this closes #5618

## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by Kayere** [comment]
Plagiarism check done ✅

**Comment by nellyat** [comment]
@mbaabuones Any updates on this PR?

**Comment by collins-a** [comment]
> @mbaabuones Any updates on this PR?

Hello @nellyat, your review shall be completed today

File: content/articles/why-devsecops-need-security-as-code/index.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+### How security as code addresses risks in development and runtime
+
+Security as code has become a driving force for the evolution of software security. Many businesses investing in data security have found security as code crucial in implementing data protection measures. In particular, security as code is helping IT firms venturing into innovation to address their security challenges better than in the past.
+
+Security as code is critical in helping organizations strengthen their software development life cycle by shifting attention to exploitable areas. Security as code automates meaningful pass or fail conditions in data pipelines. For instance, they could be set to break the software development process if it contains high-risk findings or is highly likely to expose sensitive data. This automation makes software development swift because organizations can quickly identify and fix security threats.
```
Comment (collins-a) [inline content/articles/why-devsecops-need-security-as-code/index.md:52]: All these consequent paragraphs start with _security as code_, making the tone a bit monotonous. Consider rephrasing a couple of these sentences to make them a bit more varied.
For instance, the last paragraph among the highlighted has the first two sentences starting with _security as code_, giving a rather repetitive vibe.

Also, consider going through the whole article to tweak the sections that would benefit from the above suggestion.

**Review by collins-a — CHANGES_REQUESTED** [review]
Hello @nellyat, I have enjoyed reading your article. However, you'll need to make a couple of minor modifications before proceeding to the final review. 
Additionally, going forward consider adding a _further reading/ references_ section to encourage readers to delve deeper into the subject of focus.

Ping me when done. Good luck

**Comment by nellyat** [comment]
@collins-a I have made changes to this article.

**Review by collins-a — APPROVED** [review]
Well done @nellyat, I'm satisfied with the changes

**Pull request merged**